src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputModelTypeConverter.cs

```diff
@@ -51,17 +51,27 @@ public static InputModelType CreateModelType(ref Utf8JsonReader reader, string?
                     || reader.TryReadString(nameof(InputModelType.Description), ref description)
                     || reader.TryReadString(nameof(InputModelType.Usage), ref usageString)
                     || reader.TryReadString(nameof(InputModelType.DiscriminatorPropertyName), ref discriminatorPropertyName)
-                    || reader.TryReadString(nameof(InputModelType.DiscriminatorValue), ref discriminatorValue)
-                    || reader.TryReadWithConverter(nameof(InputModelType.BaseModel), options, ref baseModel);
+                    || reader.TryReadString(nameof(InputModelType.DiscriminatorValue), ref discriminatorValue);
 
                 if (isKnownProperty)
                 {
                     continue;
                 }
-
-                if (reader.GetString() == nameof(InputModelType.Properties))
+                if (reader.GetString() == nameof(InputModelType.BaseModel))
```

Comment:
we should add some comments here to emphasize the last two properties in the JSON must be `BaseModel` and `Properties` and must in this order

Reply:
added.